Magento_Cms/templates/custom-country-form.phtml

```diff
@@ -0,0 +1,25 @@
+<?php
+/** @var \Magento\Framework\View\Element\Template $block */
+/** @var \Magento\Framework\Escaper $escaper */
+?>
+
+<h3>Search Country</h3>
+<p id="error-message"></p>
+<form id="search-country-form" data-mage-init='{"validation": {}}'>
+    <label for="country-id">Country ID</label>
+    <input type="text" name="country-id" id="country-id" class="country-id" data-validate="{'validate-country-id':true}" placeholder="Country ID..."/>
+    <button type="submit">Search</button>
+</form>
+
+<div>
+    <p id="country-name"></p>
+</div>
+
+
+<script type="text/x-magento-init">
+    {
+        "#search-country-form": {
```

Comment:
when you don't have provided options to your script better to use inline attribute instead of the `script` tag:
```html
<form data-mage-init='{"validation": {}, "Magento_Cms/js/custom-country-form": {}}'></form>
```